staging/src/k8s.io/apiserver/pkg/endpoints/handlers/response.go

```diff
@@ -35,9 +35,12 @@ import (
 // Will write the complete response object.
 func transformResponseObject(ctx context.Context, scope RequestScope, req *http.Request, w http.ResponseWriter, statusCode int, result runtime.Object) {
 	// TODO: fetch the media type much earlier in request processing and pass it into this method.
+	trace := scope.Trace
+	trace.Step("Negotiating output media type")
 	mediaType, _, err := negotiation.NegotiateOutputMediaType(req, scope.Serializer, &scope)
 	if err != nil {
 		status := responsewriters.ErrorToAPIStatus(err)
+		trace.Step("Writing raw JSON response")
```

Comment:
I think this one won't provide much value - negotiation is really fast.

Reply:
Granted but I suspect that some of the time the problem is actually writing the response to the requester. As such I would like a tight bracket around the "responsewriters.WriteRawJSON(int(status.Code), status, w)" call.

Reply:
Yes - writing response is many times dominating factor - I completely agree that.

I completely wasn't clear what I wanted to say - sorry for that.
My point was that - the time between "negotiating output media type" and this step should be negligible.

So given that your point is completely correct, I think what I wanted to say is to:
- leave this one (to see whether we spent time here)
- but remove the "negotiating output media type" step (since different between this one and that should be negligible).

Does it make sense now?

Reply:
Done